lib/client/overlays/progress.js

```diff
@@ -140,8 +156,19 @@ const init = (options, socket) => {
     // we can safely call this even if it doesn't have the class
     svg.classList.remove(`${ns}-hidden`);
 
+    if (resetTimer) {
```

Comment:
what's the reason for tracking and clearing the timeout? code comments here would be good

Reply:
There's a small chance a new HMR update becomes available when we're pending on the reset for the previous HMR update. I want to cancel the pending reset if this happens. I'll add some comments in code.